logstreams/src/main/java/io/camunda/zeebe/logstreams/impl/serializer/SequencedBatchSerializer.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.logstreams.impl.serializer;
+
+import io.camunda.zeebe.logstreams.impl.log.Sequencer.SequencedBatch;
+import io.camunda.zeebe.logstreams.log.LogAppendEntry;
+import io.camunda.zeebe.protocol.Protocol;
+import io.camunda.zeebe.scheduler.clock.ActorClock;
+import java.nio.ByteBuffer;
+import java.util.Objects;
+import org.agrona.concurrent.UnsafeBuffer;
+
+/** Accepts sequenced records, and serializes them with legacy dispatcher framing and alignment. */
+public final class SequencedBatchSerializer {
+  final LogAppendEntrySerializer entrySerializer = new LogAppendEntrySerializer();
+
+  public ByteBuffer serializeBatch(final SequencedBatch batch) {
+    Objects.requireNonNull(batch, "must provide a batch to serialize");
+
+    final var size = calculateBatchSize(batch);
+    final var buffer = ByteBuffer.allocate(size).order(Protocol.ENDIANNESS);
+    final var mutableBuffer = new UnsafeBuffer(buffer);
+
+    int bufferOffset = 0;
+    for (int i = 0; i < batch.entries().size(); i++) {
+      final var entry = batch.entries().get(i);
+      final var framedLength =
+          entrySerializer.serialize(
+              mutableBuffer,
+              bufferOffset,
+              entry,
+              batch.firstPosition() + i,
+              getSourcePosition(batch, i, entry),
+              ActorClock.currentTimeMillis());
+      // Align the next entry like the dispatcher did
```

Comment:
:wrench: Don't thin this comment is necessary, the header already explains why we're framing/aligning as the old dispatcher. If we want to make it more explicit, then I would propose explaining the logic there (e.g. "The batch is serialized first by writing out each entry, framed and aligned, and then aligning the final result. This is required for backwards compatibility with the old Dispatcher module.").